packages/model-viewer/src/test/features/scene-graph/model-spec.ts

```diff
@@ -166,6 +166,74 @@ suite('scene-graph/model', () => {
             expect(material).to.be.null;
           });
 
+      suite('Create Variant', () => {
+        test(
+            `createVariantFromMaterial() adds new primitive variants mapping
+            only to primitives that use the source material`,
+            async () => {
+              await loadModel(CUBES_GLTF_PATH);
+
+              const primitive1 = model[$primitivesList].find(prim => {
+                return prim.mesh.name === 'Box';
+              })!;
+              const primitive2 = model[$primitivesList].find(prim => {
+                return prim.mesh.name === 'Box_1';
+              })!;
+
+              const startingSize = primitive1.variantInfo.size;
+              const startingSize2 = primitive2.variantInfo.size;
+
+              // Creates a variant from material 0.
+              expect(model.createVariantFromMaterial(
+                         0, 'test-material', 'test-variant'))
+                  .to.be.ok;
+
+              // primitive1 uses material '0' so it should have a vew variant.
+              expect(primitive1.variantInfo.size).to.equal(startingSize + 1);
+              // primitive2 to should remain unchanged.
+              expect(primitive2.variantInfo.size).to.equal(startingSize2);
+            });
+
+        test('Create variant and switch to it', async () => {
+          await loadModel(CUBES_GLTF_PATH);
+
+          const primitive = model[$primitivesList].find(prim => {
+            return prim.mesh.name === 'Box';
+          })!;
+
+          model.createVariantFromMaterial(0, 'test-material', 'test-variant');
+
+          const onVariantApplied = async () => {
+            expect(primitive.getActiveMaterial().name)
+                .to.equal('test-material');
+          };
+
+          element.addEventListener('variant-applied', onVariantApplied);
+          element.variantName = 'test-variant';
+
+          await element.updateComplete;
+          element.removeEventListener('variant-applied', onVariantApplied);
```

Comment:
I think this test will pass if `variant-applied` never fires?

Reply:
It looks `await element.updateComplete;` does what I want, which is to make this behave synchronously.